single-line-diagram-server/src/main/java/com/powsybl/single/line/diagram/server/SingleLineDiagramService.java

```diff
@@ -0,0 +1,174 @@
+package com.powsybl.single.line.diagram.server;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.VoltageLevel;
+import com.powsybl.network.store.client.NetworkStoreService;
+import com.powsybl.sld.GraphBuilder;
+import com.powsybl.sld.NetworkGraphBuilder;
+import com.powsybl.sld.VoltageLevelDiagram;
+import com.powsybl.sld.layout.LayoutParameters;
+import com.powsybl.sld.layout.SmartVoltageLevelLayoutFactory;
+import com.powsybl.sld.layout.VoltageLevelLayoutFactory;
+import com.powsybl.sld.library.ResourcesComponentLibrary;
+import com.powsybl.sld.svg.DefaultDiagramInitialValueProvider;
+import com.powsybl.sld.svg.DefaultDiagramStyleProvider;
+import com.powsybl.sld.svg.DefaultNodeLabelConfiguration;
+import com.powsybl.sld.svg.DefaultSVGWriter;
+import org.apache.commons.io.IOUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody;
+
+import java.io.*;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+@ComponentScan(basePackageClasses = {NetworkStoreService.class})
+@Service
+class SingleLineDiagramService {
+
+    private static final ResourcesComponentLibrary COMPONENT_LIBRARY = new ResourcesComponentLibrary("/ConvergenceLibrary");
+    private static final LayoutParameters LAYOUT_PARAMETERS = new LayoutParameters();
+
+    @Autowired
+    private NetworkStoreService networkStoreService;
+
+    private static VoltageLevelDiagram createVoltageLevelDiagram(Network network, String voltageLevelId) {
+        VoltageLevel voltageLevel = network.getVoltageLevel(voltageLevelId);
+        if (voltageLevel == null) {
+            throw new ResponseStatusException(HttpStatus.NO_CONTENT, "Voltage level " + voltageLevelId + " not found");
+        }
+        VoltageLevelLayoutFactory voltageLevelLayoutFactory = new SmartVoltageLevelLayoutFactory(network);
+        GraphBuilder graphBuilder = new NetworkGraphBuilder(network);
+        return VoltageLevelDiagram.build(graphBuilder, voltageLevelId, voltageLevelLayoutFactory, false, false);
+    }
+
+    private Network getNetwork(String networkId) {
+        try {
+            return networkStoreService.getNetwork(networkId);
+        } catch (PowsyblException e) {
+            throw new ResponseStatusException(HttpStatus.NO_CONTENT, "Network '" + networkId + "' not found");
+        }
+    }
+
+    ResponseEntity<StreamingResponseBody> getVoltageLevelSvg(String networkId, String voltageLevelId) {
```

Comment:
A general comment for this class. Maybe we could refactor to share more code between the 3 methods  like the diagram generation part?